File: CWM.Model/Models/Searches/PartWorkOrderSearch.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Core.Models.Searches
{
    public class PartWorkOrderSearch : BaseSearch
    {
        public string? PartName { get; set; }
        public bool IncludeWorkOrder { get; set; }
        public bool IncludePart { get; set; }
        public int? UserId { get; set; }
        public int? VehicleId { get; set; }
        public int? WorkOrderId { get; set; }
        public DateTime? ServiceDate { get; set; }

    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/searches/part_work_order_search.dart
import 'package:json_annotation/json_annotation.dart';

import 'base_search.dart';

part 'part_work_order_search.g.dart';

@JsonSerializable()
class PartWorkOrderSearch extends BaseSearch {
  bool includeWorkOrder = false;
  bool includePart = false;
  int? userId;
  int? vehicleId;
  int? workOrderId;
  DateTime? serviceDate;

  PartWorkOrderSearch();

  factory PartWorkOrderSearch.fromJson(Map<String, dynamic> json) =>
      _$PartWorkOrderSearchFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PartWorkOrderSearchToJson(this);
}
